components/date-time/time.js

```diff
@@ -93,6 +105,48 @@ class TimePicker extends Component {
 		onChange( formattedDate );
 	}
 
+	updateDay() {
+		const { onChange } = this.props;
+		const { day, date } = this.state;
+		const value = parseInt( day, 10 );
+		if ( ! isInteger( value ) || value < 1 || value > 31 ) {
+			this.syncState( this.props );
+			return;
+		}
+		const newDate = date.clone().date( value );
+		this.setState( { date: newDate } );
+		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
+		onChange( formattedDate );
+	}
+
+	updateMonth() {
```

Comment:
The `update[Day/Month/Year]` methods are all very similar to each other, save the `newDate` and `value` checks. Could they be DRY'd a bit so they duplicate less code?

Reply:
True. I felt I was following some convention by doing it this way. Sometimes too much metaprogramming results in less readability too.

Reply:
Yeah, definitely. Readable repetition over clever metaprogramming anyday 👍

*But* **if** it could be tightened up that'd be cool. 😉